evap/staff/views.py

```diff
@@ -422,6 +438,8 @@ def semester_delete(request):
 
     if not semester.can_be_deleted_by_manager:
         raise SuspiciousOperation("Deleting semester not allowed")
+    if semester.is_active:
```

Comment:
Should be contained in `can_be_deleted_by_manager`

(besides the semantic reasoning, here you also have the safety concern: you could have easily forgotten the check here)